cart/views.py

```diff
@@ -6,18 +6,154 @@
 
 from cart.models    import Cart, Order, OrderStatus, AddressInformation
 from user.models    import User
+from user.utils     import login_decorator
 from product.models import Product
 from voucher.models import Voucher
 
 
-class UserCartView(View):
+ORDER_STATUS_CHECK = 'checking_out'
+
+
+class CartProductView(View):
     @login_decorator
     def post(self, request):
-        data       = json.loads(request.body)
-        user_id    = request.user.id
-
-        Cart.objects.create(
-                product_id = 
-                user_id    = user_id,
-                voucher_id = voucher_id
-                )
+        try:
+            data             = json.loads(request.body)
+            product_id       = data.get('product_id')
+            user_id          = request.user.id
+
+            if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK, carts__product_id=product_id).exists():
+
+                return JsonResponse({'message': False}, status=403)
+
+            if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).exists() and not Cart.objects.filter(product_id=product_id, order__order_status__name=ORDER_STATUS_CHECK, order__user_id=user_id).exists():
```

Comment:
if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).exists()
=> order에 order_status가 chacking_out 
위의 조건 너무 잘 만드셨습니다.
저희 원하는 조건은 로그인 한 유저의 order 상태가 checking_out이고(위의 조건) 그 때 연결된 cart에 product_id가 없을 때 인데, 이미 맨 처음 if에서 product_id가 있는 조건을 지나쳐 왔기 때문에 and 이후의 조건은 필요 없을 것 같습니다.
그런데 이것은 어디까지나 생각일 뿐입니다. 실제 shell에서 어떻게 나오는지 확인해 볼 필요가 있습니다.
```suggestion
            if Order.objects.filter(user_id=user_id, order_status__name=ORDER_STATUS_CHECK).exists():
```